src/screens/TaskDetailScreen.tsx

```diff
@@ -191,255 +149,214 @@ const TaskDetailScreen: React.FC<TaskDetailScreenProps> = ({ navigation, route }
         'Are you sure you want to delete this task?',
         [
-          { text: 'Cancel' },
+          { text: 'Cancel', style: 'cancel' },
           {
             text: 'Delete',
             style: 'destructive',
             onPress: () => {
-              // Fade out animation before navigation
-              Animated.timing(fadeAnim, {
-                toValue: 0,
-                duration: 200,
-                useNativeDriver: true
-              }).start(() => {
-                onDelete(task.id);
-                navigation.goBack();
-              });
+              onDelete(task.id);
+              navigation.goBack();
             }
           }
         ]
     );
-  }, [navigation, task.id, onDelete, fadeAnim]);
+  };
 
   /**
    * Formats a date string to a more readable format
    * @param dateString - ISO date string to format
    * @returns Formatted date string (e.g., "April 19, 2025")
    */
-  const formatDate = useCallback((dateString: string) => {
+  const formatDate = (dateString: string) => {
     const date = new Date(dateString);
     return date.toLocaleDateString('en-US', {
       year: 'numeric',
       month: 'long',
       day: 'numeric'
     });
-  }, []);
-
-  // Memoize dynamically computed styles to avoid recreation on each render
-  const computedStyles = useMemo(() => ({
-    statusBadge: [
-      styles.statusBadge,
-      task.status === 'completed' ? styles.completedBadge : styles.pendingBadge
-    ],
-    statusText: [
-      styles.statusText,
-      task.status === 'completed' ? styles.completedText : styles.pendingText
-    ],
-    toggleButton: [
-      styles.button,
-      task.status === 'completed' ? styles.pendingButton : styles.completeButton
-    ]
-  }), [task.status]);
-
-  // StatusBadge component to improve readability in render function
-  const StatusBadge = () => (
-      <View style={styles.statusContainer}>
-        <View style={computedStyles.statusBadge}>
-          <Text style={computedStyles.statusText}>
-            {task.status === 'completed' ? 'Completed' : 'Pending'}
-          </Text>
-        </View>
-      </View>
-  );
+  };
 
   return (
-      <>
-        <StatusBar barStyle="dark-content" />
-        <Animated.ScrollView
-            style={[styles.container, { opacity: fadeAnim }]}
-            showsVerticalScrollIndicator={false}
-        >
-          {/* Task details card */}
-          <View style={styles.card}>
-            <Text style={styles.title}>{task.title}</Text>
-
-            <Text style={styles.sectionTitle}>Description</Text>
-            <Text style={styles.description}>
-              {task.description || 'No description'}
-            </Text>
-
-            <Text style={styles.sectionTitle}>Created Date</Text>
-            <Text style={styles.date}>{formatDate(task.createdAt)}</Text>
-
-            <Text style={styles.sectionTitle}>Status</Text>
-            <StatusBadge />
-          </View>
+      <ScrollView style={styles.container}>
```

Comment:
Entire render (lines 190–261): The JSX is getting hefty. Spinning out:

<StatusBadge status={task.status} />

<ActionButton icon="share" label="Share" onPress={handleShare} />
will make this file more readable and easier to unit-test.